### ISSUE TITLE
In DataType Manager, it would be useful to have a method to edit TypeDefDataTypes

### ISSUE THREAD

**Issue by marcushall42** [NationalSecurityAgency/ghidra#3113]
**Is your feature request related to a problem? Please describe.**
In the DataType Manager window, if a TypeDefDataType is present, the edit action edits the Base DataType.  There seems to be no way to re-direct the TypeDef to a different type.  It would be useful if the Edit action (or a context-dependent action like "Display BaseType") was able to choose a different DataType.

**Describe the solution you'd like**
Some way to edit the base DataType of the Typedef.

**Describe alternatives you've considered**
At the moment, the only way seems to be to create another Typedef, replace the typedef, and re-name.  Although this results in any uses of the datatype being marked as changed.  From a script, I can create an object that implements TypeDef and replace the contents with that, but the GUI doesn't seem to have any way to do this.

**Comment by dragonmacher** [comment]
> Although this results in any uses of the datatype being marked as changed. 

If you are changing the type def, then that implies all uses of the type def need to be updated.   If you do not want the existing application of that typ def to change, then that implies you are creating a new type def, which would allow the existing uses to use the old version.

We could create an action in the UI to `Edit -> Base type...`.   This action would function under the hood like the existing case you mentioned above: 
> At the moment, the only way seems to be to create another Typedef, replace the typedef, and re-name

Is an action such as this what you are requesting?

**Comment by Wall-AF** [comment]
If @marcushall42 isn't, I'd like it!

**Comment by marcushall42** [comment]
Yes, in the UI being able to edit the typedef data type similar to editing a structure or function definition would be great.  I would expect it to pop up a datatype chooser window similar to the dialog that initially defines the typedef.

In my case, I have a large number of related programs in the project and there are structures that evolve with different versions of the source program.  So, I have a structure like:
  taskDef               typedef pointing to taskDef_local
  taskDef_local     typedef pointing to taskDef_V0
All of the general pointers point to the taskDef typedef datatype, so function parameters, etc. all use that.  The taskDef and taskDef_V0, taskDef_V1, etc. all are in a datatype archive, so they are shared between the various programs.  So, if a new program adds some data to the taskDef structure, I define a new taskDef_V7 (or whatever) and point taskDef_local to that.  But, there isn't any intuitive way to edit taskDef_local to now point to taskDef_V7.  It turns out that if taskDef_local currently points to taskDev_V0, then drag/drop of taskDef_V7 onto taskDef_V0 accomplishes what I want, but it isn't very intuitive to ask users to do..

I do have a script to edit the variant datatypes, so this is no longer a high priority for me.  And it turns out that trying to use DataTypeChooser with a filter text is near impossible (hence issue 3140!)

**Comment by Wall-AF** [comment]
My use case is that when I'm using the `Parse C Source...` I have 16- & 32-bit pointers, but the parser doesn't recognise the `FAR`/`far` (old tech) keyword.  Meaning all the imported typedef's must be corrected before they can be used in anger!

Any suggestions of how to _fix_ the parser are gratefully received!

**Comment by Wall-AF** [comment]
> My use case is that when I'm using the `Parse C Source...` I have 16- & 32-bit pointers, but the parser doesn't recognise the `FAR`/`far` (old tech) keyword. Meaning all the imported typedef's must be corrected before they can be used in anger!
> 
> Any suggestions of how to _fix_ the parser are gratefully received!

For someone in my situation, [here's](https://github.com/user-attachments/files/17438281/_ReplaceTypedefPtrWithFarPtrScript.txt) an updated version of @astrelsky's script (from #4544) that'll replace all typedef's selected in the `Data Type Manager` (providing they are pointer one's) with sized pointer of 32-bits.
